Multiplayer basic functionality

ThatOneGuySam opened the pull request.

ThatOneGuySam committed:
Countdown to game
Now when the max players have joined a lobby, a countdown runs before putting them all in a game. Next steps is to display the countdown to users, make sure it resets when someone disconnects, and ustilize the data sent when they start to actually keep everyone connected.

ThatOneGuySam committed:
Countdown displayed
Obviously better design needed, but I just used the messages during the counter update for a bare bones counter for users to see. Still need to deal with disconnects and actually utilize sent data now

ThatOneGuySam committed:
Multiplayer movement!!!
The socket now is passed to the game, and the simple instance of sending and receiving messages on players moving works! Next up is making more complicated messaging for player data, but this is very very useful!

aidandoherty01 reviewed (approved):
Look good, only question is should this be pushed if we plan on making large changes to **lobbyClass.js**?

ThatOneGuySam committed:
gameSession Class
Relatively little has chanegd functionality-wise, but now everything's stored in one game object for ease. Something to be careful about is javascript likes to dynamically change types while passing values around, so be wary (I had a two hour bug that was it was converting a user defiend class to generic object)

aidandoherty01 reviewed (approved):
Great work dude 👍
